src/cart/crt_register.c

```diff
@@ -617,11 +617,11 @@ crt_proto_query(crt_endpoint_t *tgt_ep, crt_opcode_t base_opc,
 
 	D_ALLOC_PTR(proto_query);
 	if (proto_query == NULL)
-		return -DER_NOMEM;
+		D_GOTO(out, rc = -DER_NOMEM);
 
 	D_ALLOC_PTR(proto_query->pq_coq);
 	if (proto_query->pq_coq == NULL)
-		return -DER_NOMEM;
+		D_GOTO(out, rc = -DER_NOMEM);
```

Comment:
I spotted a few in this code (although this one was me, it only landed last week) so i was going to try it though the fault injection framework next week.